### DISCUSSION TITLE
I would like to work on the Treemaps GSOC project this summer!

### DISCUSSION THREAD

**Discussion by MahmoudHamdy02** [owid/owid-grapher#1549] [General]
Hello OWID team!  

I would love to work on the [treemaps project](https://github.com/owid/owid-grapher/discussions/1287) throughout this summer, I've done a lot of research on the topic and laid out a plan for completing it in [my proposal](https://docs.google.com/document/d/1iCEFWJVLfXODaqt4XTSumu7V3uZpy6fsFaPALfnjyYM/edit?usp=sharing). 

Since the project doesn't need to fit in a 12 week schedule, the timeline can be more flexible if needed.

The possible mentors for this project were @danyx23 @danielgavrilov @larsyencken, please let me know your opinion!

**Comment by danielgavrilov** [comment]
Hi @MahmoudHamdy02, thanks a lot for your contributions! 

Lars and Daniel are away this week so we will be a bit slow responding, but you should get a response next week.

On my side: It would be awesome to work on this, but I'm not sure if we can offer our own internships yet (the GSoC internships are set up and funded by Google). On mentorship, I think we should have the capacity, but have to check with the others once they're back.

**Reply by MahmoudHamdy02** [reply]
Hello @danielgavrilov!

I was wondering if there were any updates on this, thanks!

**Reply by danielgavrilov** [reply]
Hi Mahmoud, sorry for the slow reply.

We won’t be able to do a paid internship unfortunately, so it would only be possible to contribute as a volunteer. 

If you are interested in that, let me know. I am happy to help you get started if you have questions on how to approach it. 

(I've been slow to respond here, and quite slow to review your latest PR – sorry about that. There's a project that has been consuming a lot of my time recently, I'll likely be more available in ~2 weeks.)

**Comment by MahmoudHamdy02** [comment]
Hi Daniel, 

That wouldn't be a problem, I'd love to contribute to gain some work experience!

I'm ready to start when possible, and I'd like to know more details about the initial approach.

I have a few questions about the workflow:
- Would there be one or multiple mentors to follow my work progress? 
- What would be a suitable place in case I have some technical questions?
- Would would be the general timeline for this project, as I would be working as a volunteer? 

I'm looking forward to hearing from you!

Thanks a lot!

**Reply by danielgavrilov** [reply]
On workflow:

- It might be mostly me as a mentor for now, but there will be someone else in case I'm away.
- I have invited you to a channel on our Slack, you'll get the fastest answer there. Also feel free to use GitHub issues/discussions, they are nicer for longer writeups, and they have the benefit of being open to everyone. (But it's still standard team practice to post the discussion link on Slack, since there are a lot of GitHub notifications, it's easy to miss.)
- Since you'd be volunteering, the timeline is very much up to you. You can spend more time when you are free, less when you are not. You can spend more time when something is interesting and where you feel you are learning (e.g. maybe trying out treemap algorithms). On our side, we don't have this project on our roadmap for the year, so there is no rush to get it done. Even though there are some charts we'd like to have as treemaps, there are other projects we wanted to get done first. However, if you have some time constraints you'd like to establish, we can help manage that.

About the approach:
- We need to discuss how the chart configuration could work to support treemaps. It's an unusual chart, the Grapher doesn't really have any hierarchical chart like this. I have some ideas but will think about it a bit more.
- To start, we also need to set up all the boilerplate stuff, like creating a new chart type and implementing the interface Grapher expects. 

Let's maybe set up a time to discuss this – if you are happy to join Slack, I'll get in touch there.

**Reply by MahmoudHamdy02** [reply]
Thank you for the detailed response, I really appreciate it! 

I have joined the slack channel, we can start whenever it's convenient for you to do so.

Have a great day!
